content/en/blog/performance-benchmarks-gke.md

```diff
@@ -0,0 +1,354 @@
+
+   1. Pods belonging to a completed LoadTest are in terminated state, and so
+      consume no resources on the cluster. However, terminated pods can be
+      garbage-collected at any time.
+
+   1. If we let pods belonging to all completed tests stay in our continuous
+      integration cluster, we find that they are garbage-collected within one
+      hour.
+
+   1. If we delete LoadTest resources for tests that complete successfully, the
+      associated pods are also deleted. In this case, the pods belonging to
+      _unsuccessful_ tests, which are few in number and may be useful for
+      debugging, stay on the cluster until the 24h TTL is reached.
+
+[test runner]:
+  https://github.com/grpc/test-infra/blob/master/tools/README.md#test-runner
+
+### Dashboard
+
+Test results from continuous integration are saved to [BigQuery]. The data
+stored in BigQuery is then replicated to a Postgres database for visualization
+on a dashboard.
+
+The code for the dashboard, as well as the configuration of the main continuous
+integration dashboard, are stored in the [Test Infra] repository. This brings
+the following benefits:
+
+1. The main dashboard is maintained by updating the stored configuration,
+   instead of updating it directly in the UI.
+
+1. Users can deploy their own dashboard, using their own configuration.
+
+This contrasts with the dashboard for the previous benchmarks, built using
+Perfkit Explorer, which was maintained by updating it directly in the UI, and
+could not be easily replicated by users.
+
+For details, see [dashboard implementation].
+
+Dashboard snapshot:
+
+[![Dashboard snapshot]](https://grafana-dot-grpc-testing.appspot.com)
+
+[bigquery]: https://cloud.google.com/bigquery
+[dashboard implementation]:
+  https://github.com/grpc/test-infra/tree/master/dashboard/README.md
+[dashboard snapshot]: /img/blog/performance-benchmarks-gke/dashboard.png
+
+## Results
+
+The following observations can be made on the results and user experience of the
+gRPC benchmarking on GKE:
+
+1. Performance metrics (latency, QPS, etc.) produce the same or better results
+   as the old benchmarks on GCE.
+
+1. The overhead of pod creation and deletion for each test in GKE is small (less
+   than 15s) in our benchmarking cluster.
+
+1. Test images are dockerized and started again for each test, resulting in
+   several benefits:
+
+   1. Results are more consistent.
+
+   1. Runtime errors are rare.
+
+   1. System is split into well-defined components, resulting in simpler
+      upgrades.
+
+   1. Tests can be easily parallelized, resulting in faster execution times.
+
+   1. Experiments are easier to perform.
+
+Examples of best practices and insights derived from experimentation:
+
+1. Use `c2` instances for clients and servers (instance types matter a lot for
+   the observed latency and its variance and also for the measured throughput).
+
+1. For Java under Docker, one must make sure to set CPU limits (this might also
```

Comment:
This item (2) is less important than (3) so pehaps we can list it as 3.

Also without context, "one must make sure to set CPU limits" is not understandable. 
Include more detail, e.g. that under docker containers java might not be able to detect the number of available processors, which can then give very pessimistic results since grpc-java uses the detected number of processor for sizing the thread pools for processing events etc.

Reply:
Reordered and updated the item.